python/ray/train/_internal/dataset_spec.py

```diff
@@ -213,6 +221,9 @@ def get_dataset_shards(
             else:
                 dataset_splits = [dataset] * len(training_worker_handles)
 
+            for i, dataset_split in enumerate(dataset_splits):
+                dataset_splits[i] = dataset_split.iterator()
```

Comment:
return type hint for this method should be changed to `Dict[str, DatasetIterator]`